client/src/components/dashboard/DashboardCanvasItem.jsx

```diff
@@ -28,12 +28,33 @@ export default class DashboardCanvasItem extends Component {
     }, 0);
   }
 
-  getItemLayout() {
+  shouldComponentUpdate(nextProps) {
+    const oldLayout = this.getItemLayout(this.props);
+    const newLayout = this.getItemLayout(nextProps);
+    const layoutsExist = Boolean(oldLayout && newLayout);
+    const dimensionsUnchanged = layoutsExist ?
+      oldLayout.w === newLayout.w && oldLayout.h === newLayout.h : false;
+    const canvasWidthUnchanged = this.props.canvasWidth === nextProps.canvasWidth;
+    const needDataset = this.props.item.type === 'visualisation';
+    const datasetDependencyMet = needDataset ? this.getIsDatasetLoaded(this.props) : true;
+    const styleTransitionFinished = this.state.style.boxShadow === 'none';
+
+    const shouldNotUpdate = Boolean(
+        dimensionsUnchanged &&
+        canvasWidthUnchanged &&
+        datasetDependencyMet &&
+        styleTransitionFinished
+      );
+
+    return !shouldNotUpdate;
```

Comment:
Yeah, this felt a bit funny to me too as I was writing it, but in the end I decided it was better to have one negation in the return statement than multiple in the test. But coming back to it, I suppose it might be clearer like:

````
+    const shouldUpdate = Boolean(
+        dimensionsChanged ||
+        canvasWidthChanged ||
+        !datasetDependencyMet ||
+        !styleTransitionFinished
+      );
+
+    return shouldUpdate;
````
I would be reluctant to change e.g. `datasetDependencyMet` to `datasetDependencyNotMet` to further avoid negation as that feels even more confusing to me

What do you think?